### DISCUSSION TITLE
Updating Maven Surefire version in Tycho

### DISCUSSION THREAD

**Discussion by joeshannon** [eclipse-tycho/tycho#751] [Ideas]
The current version used is 2.22.2.

To facilitate migration to JUnit 5, it would be a great benefit if display names were supported (#198/#381). As mentioned in those issues, this requires a newer version of Surefire.

Is this something that is suitable to start working on now? One consideration is that Surefire is still released as a milestone e.g. the current version is 3.0.0-M5 (released in June 2020). I can't see a documented timescale for a 3.0.0 release but there is some outline of further milestone releases on the [plugin docs](https://maven.apache.org/surefire/maven-surefire-plugin/).

I've had a go at updating this to see what happens. I can get Tycho to compile and build but think some deeper work will be required due to reporting api changes.

**Comment by mickaelistria** [comment]
IMO 3.0.0.M5 is now stable and mainstream enough to start relying on it.
@laeubi @kwin I know you guys are following Maven development more closely, do you have an insights on that topic; or maybe even a way to influence in favor of an upcoming 3.0.0 release soon?

**Reply by laeubi** [reply]
I think one could simply ask for a release, but I don't think we need to wait for one technically. I see mayn projects simply using the 3.0.0.M5 in their poms anyways.

**Comment by mickaelistria** [comment]
FWIW, I'm currently looking at adopting surefire 3.0.0-M5 in Tycho. See progress at #802
